alacritty_terminal/src/term/mod.rs

```diff
@@ -1256,10 +1268,55 @@ impl<T> Term<T> {
         self.event_proxy.send_event(Event::Exit);
     }
 
+    #[inline]
     pub fn clipboard(&mut self) -> &mut Clipboard {
         &mut self.clipboard
     }
 
+    /// Toggle the keyboard motion mode.
+    #[inline]
+    pub fn toggle_keyboard_mode(&mut self) {
+        self.mode ^= TermMode::KEYBOARD_MOTION;
+        self.grid.selection = None;
+
+        // Reset keyboard cursor position to match primary cursor
+        if self.mode.contains(TermMode::KEYBOARD_MOTION) {
+            let line = min(self.cursor.point.line + self.grid.display_offset(), self.lines() - 1);
+            self.keyboard_cursor = KeyboardCursor::new(Point::new(line, self.cursor.point.col));
+        }
+
+        self.dirty = true;
+    }
+
+    /// Move keyboard motion cursor.
+    #[inline]
+    pub fn keyboard_motion(&mut self, motion: KeyboardMotion)
+    where
+        T: EventListener,
+    {
+        self.dirty = true;
```

Comment:
Could we place it in the end of function?

Reply:
We can't, because we early return.